### PR TITLE
Matmul Precision issue due to fused multiply add: fixed by increasing RTOL to pass over 9 orders of magnitude for input ranges.

### PR DESCRIPTION
**Pull request by AlexandreEichenberger**
There was an issue with numerical tests of matmul and gemm with random number beyond the -1.0 to 1.0 range.

Testing with disabling the fused multiply-add succeeded when performing 10,000 tests using the tight relative tolerance of 1e-5. When enabling fused multiply add (FMA), random coefficients in the range below passed the test with the following  in relative tolerances
```
+- 10 passed with rtol 3e-5
+- 1k passed with rtol 5e-5
+- 100k passed with rtol 5e-5
+=100M passed with rtol 3e-5
```

As a result, I did set conservatively the relative tolerance to 5e-5, which passes tests within 9 orders of magnitudes for the input coefficients.

While it is not necessary to set the range of random numbers within large numbers, for sanity I let the TestMatMul2D use values in the -10.0 to +10.0 range.

### CODE REVIEW

**Comment by AlexandreEichenberger** [comment]
associated issue: #1286

File: test/modellib/ModelLib.cpp [hunk 1]
```diff
@@ -134,7 +134,11 @@ bool ModelLibBuilder::areCloseFloat(
     const OMTensor *res, const OMTensor *ref) const {
   if (!res || !ref)
     return false;
-  float rtol = getenv("TEST_RTOL") ? atof(getenv("TEST_RTOL")) : 1e-5;
+  // Because of floating point impact of using FMA instead of multiply followed
+  // by an add,the default RTOL is increased from 1e-5 to 5e-5. This bound has
```
Comment (etiotto) [inline test/modellib/ModelLib.cpp:138]: [nit] missing space  "add,the" -> "add, the"

**Review by etiotto — APPROVED** [review]
LGTM.

File: test/modellib/ModelLib.cpp (hunk 1 above)
Comment (AlexandreEichenberger) [reply test/modellib/ModelLib.cpp:138]: tx

**Comment by imaihal** [comment]
@AlexandreEichenberger Thanks for the investigation. This kinds of work should take much time... 

It seems the Jenkins test didn't pass. Should we change RTOL and ATOL?

```
a[10,0] = -3.22827 != b[10,0] = -3.22778
```
I think this difference is bit large, but do you think this difference is reasonable when using FMA ?

**Comment by AlexandreEichenberger** [comment]
> I think this difference is bit large, 

I agree, and the results from the CIs don't seem to agree with the experiments that I did running the tests on Z14 and x86 Linux. Will investigate some more.

**Pull request closed without merging**